libnetdata/libnetdata.h

```diff
@@ -319,7 +319,7 @@ extern char *netdata_configured_host_prefix;
 #include "log/log.h"
 #include "procfile/procfile.h"
 #include "dictionary/dictionary.h"
-#ifdef HAVE_LIBBPF
+#if defined(HAVE_LIBBPF) && !defined(__cplusplus)
```

Comment:
Please, can  you give us some context for this change? As you can see [here](https://elixir.bootlin.com/linux/latest/source/tools/lib/bpf/btf.h#L14) the `libbpf ` is prepared to be used with `C++`.

Reply:
```
In file included from ./daemon/../libnetdata/ebpf/ebpf.h:6,
                 from ./daemon/../libnetdata/libnetdata.h:323,
                 from ./daemon/common.h:6,
                 from ml/ml-private.h:16,
                 from ml/ml_train.cc:1:
externaldeps/libbpf/include/bpf/bpf.h:237:6: error: use of enum ‘bpf_stats_type’ without previous declaration
  237 | enum bpf_stats_type; /* defined in up-to-date linux/bpf.h */
      |      ^~~~~~~~~~~~~~
```

```
(v3.9.1) vk@nx9420:~$ grep -RIn 'bpf_stats_type' /usr/include/linux/ | wc -l
0
(v3.9.1) vk@nx9420:~/repos/nd/prs/cxx$ lsb_release -a 
No LSB modules are available.
Distributor ID: Linuxmint
Description:    Linux Mint 20
Release:        20
Codename:       ulyana
```

Reply:
Thank you for the context, I can understand better now what is happening on your distribution.

Reply:
i think better to solve at the source instead of skipping header?